internal/clusterfeature/interface.go

```diff
@@ -93,12 +93,14 @@ type FeatureManager interface {
 
 	// Name returns the feature's name.
 	Name() string
+
+	BeforeSave(ctx context.Context, clusterID uint, spec FeatureSpec) (FeatureSpec, error)
 }
 
 // FeatureOutputProducer defines how to produce a cluster feature's output.
 type FeatureOutputProducer interface {
 	// GetOutput returns a cluster feature's output.
-	GetOutput(ctx context.Context, clusterID uint) (FeatureOutput, error)
+	GetOutput(ctx context.Context, clusterID uint, spec FeatureSpec) (FeatureOutput, error)
```

Comment:
What's the point? This doesn't belong here IMHO.

Reply:
Unfortunately, we need input fields (specially `secretID` to communicate with Vault) to generate feature output

Reply:
Same as above